### DISCUSSION TITLE
Alien can still spawn "through" closed/locked/powered off vent iris

### DISCUSSION THREAD

**Discussion by ghramsey** [MattFiler/LegendPlugin#145] [General]
I setup a tinybutton entity and locked a vent.  This is in mission 3 where I also spawned multiple aliens.

Even so, with the vent iris indicating locked and power off an alien was able to spawn "through" the closed iris.
I made the silly assumption that the vent being closed would act like a closed/locked door.

I presume I need some sort of collision barrier for the alien?

Any help with this?

**Comment by ghramsey** [comment]
I ended up setting up collision barriers on the two vent entrances.  It's not ideal but it does work after a fashion.  Interesting though 
when I got to M4 the comms array control room the wall vent near the entrance is a common place for extra alien to spawn from,  
All I had to do was lock the vent cover and the aliens cannot spawn there.  No idea what the difference between the two vents is 
other than one required extra measures to keep them spawning. when it was closed.

**Reply by MattFiler** [reply]
This is interesting - perhaps something MVR related in their compiler which would set up the collision correctly which just soley editing the Commands isn't doing. I'm looking into this sort of issue currently - will get back to you!

**Reply by MattFiler** [reply]
(It's probably somewhat related to #144 and #95)

**Reply by ghramsey** [reply]
I ended up creating collision barriers in front of the vent and activating them via switch.    I also put two more in front of 
each opening of the metal detector thing near the vent.  Those work but the alien walks/runs into them blindly and just 
ends up stuck, running-in-lace.  Even if I open the vent and make myself accessible the alien doesn't stop trying to run 
through the barrier.   I am not sure what I am doing wrong there,    Do I need to use something other than a collision 
barrier for the aliens here ? Or perhaps I used the wrote type.   I used type 2 which is "generic" I think in the ENUM list.

What's also interesting and related is that I locked the iris vent in the Comms Array room in M4.  I think I merely set it to 
lock on reset (I can't find what I actually edited that triggered it to lock. I know I was messing with settings on iris vents.). 
In that case the vent IS preventing the aliens from either spawning or entering and they are correctly repulsed by it and
 only spawn from the vent in the center of the room.

So Issue #95 was also in Tech_Hub.  I wonder what's different between the two maps that the vents 
being locked have completely opposite effects?

Any help or insight you can lend will be helpful.

**Comment by ghramsey** [comment]
More experimenting and it seems the lock_on_reset boolean  is the key to the proverbial castle.  
Using lock_on_reset in the iris vent entity seems to permit locking and makes the correct barrier to entry/exit once the iris is closed.   
I don't quite understand why that would be but it sorta negates the collision barrier work arounds.  
And looking at the iris in mission 4 that is what I evidently did that made the vent close/lock and likewise prevented spawn once locked.

**Comment by ghramsey** [comment]
So, on further investigation on this the issue is that powering off /  locking a vent does not tell the alien AI that it cannot use it.

Using Force_close however does;  I've found for example setting a door to close on reset does not prevent the alien from attempting to wait for the door to open even if the door is made not to be triggered by the alien.   The alien will wait, screaming in rage that the door is not opening.  It stands there even if the player in inaccessible and making noise.  This is similar to how it passes through the closed/locked iris vent.  Something is missing when closed to inform the AI that the closed iris is also not for use and make it turn away from trying to use it.

Emergency_close / Force_close such as is used in the doors in the medical area however does trigger this essential bit.  
 The alien both turns from such a door or vent and will upvent if a player it inaccessible.
